src/lib.rs

```diff
@@ -171,7 +171,7 @@ macro_rules! trace {
                     .set_bold(true),
             )
             .unwrap();
-        writeln!(&mut stream, "[{} {}] {}", now, rall::Level::TRACE, $str).unwrap();
+        writeln!(&mut stream, "[{} {}] {}", now, rall::Level::TRACE, format!($($arg)*)).unwrap();
```

Comment:
Could this use `format_args!` instead?
Same question for the other macros.

Reply:
I sadly don't have that much experience with rust macros (yet!) 
@Allstreamer do you have any thoughts on this?

Reply:
[https://doc.rust-lang.org/std/macro.format_args.html](Refrence)
Suposedly it avoids heap allocations when dealing with Strings,
not sure either tho

Reply:
Hmm interesting, I'll run the pipeline and merge if that succeeds for now but definitely something to look into.